packages/insured-bridge-relayer/test/CrossDomainFinalizer.ts

```diff
@@ -0,0 +1,257 @@
+
+    bridgeDepositBox = await BridgeDepositBox.new(
+      l2BridgeAdminImpersonator,
+      minimumBridgingDelay,
+      ZERO_ADDRESS,
+      l2Timer.options.address
+    ).send({ from: l2Owner });
+
+    l2Token = await ERC20.new("L2ERC20", "L2ERC20", 18).send({ from: l2Owner });
+    await l2Token.methods.addMember(TokenRolesEnum.MINTER, l2Owner).send({ from: l2Owner });
+
+    await bridgeAdmin.methods
+      .setDepositContract(chainId, bridgeDepositBox.options.address, messenger.options.address)
+      .send({ from: l1Owner });
+    // New BridgePool linked to BridgeAdmin
+    bridgePool = await BridgePool.new(
+      "LP Token",
+      "LPT",
+      bridgeAdmin.options.address,
+      l1Token.options.address,
+      lpFeeRatePerSecond,
+      false,
+      l1Timer.options.address
+    ).send({ from: l1Owner });
+
+    // Add L1-L2 token mapping. Note that we need to whitelist on both the L1 and L2 side because the L1 mapping
+    // is used by the bots to fetch bridge pool addresses.
+    await bridgeAdmin.methods
+      .whitelistToken(
+        chainId,
+        l1Token.options.address,
+        l2Token.options.address,
+        bridgePool.options.address,
+        0,
+        defaultGasLimit,
+        defaultGasPrice,
+        0
+      )
+      .send({ from: l1Owner });
+
+    await bridgeDepositBox.methods
+      .whitelistToken(l1Token.options.address, l2Token.options.address, bridgePool.options.address)
+      .send({ from: l2BridgeAdminImpersonator });
+
+    spy = sinon.spy();
+    spyLogger = winston.createLogger({
+      level: "debug",
+      transports: [new SpyTransport({ level: "debug" }, { spy: spy })],
+    });
+
+    // Create the rate models for the one and only l1Token, set to the single rateModel defined in the constants.
+    const rateModels = { [l1Token.options.address]: rateModel };
+    l1Client = new InsuredBridgeL1Client(spyLogger, web3, bridgeAdmin.options.address, rateModels);
+    l2Client = new InsuredBridgeL2Client(spyLogger, web3, bridgeDepositBox.options.address, chainId);
+
+    gasEstimator = new GasEstimator(spyLogger);
+    crossDomainFinalizer = new CrossDomainFinalizer(spyLogger, gasEstimator, l1Client, l2Client, l1Relayer);
+  });
+  describe("L2->L1 cross-domain transfers over the canonical bridge", () => {
+    beforeEach(async function () {
+      // Add liquidity to the L1 pool to facilitate the relay action.
+      await l1Token.methods.mint(l1LiquidityProvider, initialPoolLiquidity).send({ from: l1Owner });
+      await l1Token.methods
+        .approve(bridgePool.options.address, initialPoolLiquidity)
+        .send({ from: l1LiquidityProvider });
+      await bridgePool.methods.addLiquidity(initialPoolLiquidity).send({ from: l1LiquidityProvider });
+
+      // Mint some tokens for the L2 depositor.
+      await l2Token.methods.mint(l2Depositor, depositAmount).send({ from: l2Owner });
+    });
+    it("Can correctly detect and cross-domain transfers", async function () {
```

Comment:
Can you add a set of test where there are two L2 whitelisted tokens:
- Both L2 tokens can be bridged
- Only one can be bridged because `canBridge` returns false for one of them so only one bridging action is sent
- Both L2 tokens can be bridged but one fails to send for some reason. For example, you could whitelist one L2 token that always reverts on `safeTransferFrom`? I know this is annoying but definitely increases coverage.

Reply:
done